### PR TITLE
feat: add new content navigator component

### PR DESCRIPTION
**Pull request by SebasthianSalpozz**
#### Resolved task/US: ([item title]())
* * * *

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: package-lock.json
Comment (samuel-escalera-jalafund) [inline package-lock.json:1]: Isn't necessary upload this file, remove of the pr

File: src/components/ContentNavigator/Sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+} from "@mui/material";
+
+interface SidebarProps {
+  sections: { id: string; title: string }[];
+  onScrollToSection: (id: string) => void;
+}
+
+const Sidebar: React.FC<SidebarProps> = ({ sections, onScrollToSection }) => {
+  return (
+    <Box width="20%" bgcolor="#1a1a1a" color="white" p={2}>
```
Comment (samuel-escalera-jalafund) [inline src/components/ContentNavigator/Sidebar.tsx]: For the bgcolor, use the colors in styleCOnstant, like:
`<Typography color={COLORS.GREEN} >HOLAAAA</Typography>`

File: src/components/ContentNavigator/ContentNavigator.tsx
Comment (samuel-escalera-jalafund) [inline src/components/ContentNavigator/ContentNavigator.tsx:1]: Add an index.tsx for this component

File: src/pages/Overview/Overview.tsx
Comment (samuel-escalera-jalafund) [inline src/pages/Overview/Overview.tsx:1]: - Add an index.tsx for this page
- Change the component name Overview for a most described name, a name which indicates this is a documentation overview

File: src/pages/SetUpGradle/SetUpAgentGradle.tsx
Comment (samuel-escalera-jalafund) [inline src/pages/SetUpGradle/SetUpAgentGradle.tsx:1]: - Add an index.tsx for this page
- Change the component name for a most described name, a name which indicates this is about documentation

File: src/pages/setupMaven/SetupAgentInMaven.tsx
Comment (samuel-escalera-jalafund) [inline src/pages/setupMaven/SetupAgentInMaven.tsx:1]: - Add an index.tsx for this page
- Change the component name for a most described name, a name which indicates this is about documentation

**Review by samuel-escalera-jalafund — CHANGES_REQUESTED** [review]
For all the .png dont save that in different resource directors, put your .png inside the assets/images directory and inside this directory subdivide the images subdivide the images in subdirectory depending the component or page

File: package-lock.json
Comment (SebasthianSalpozz) [reply package-lock.json:1]: Sure

File: src/components/ContentNavigator/Sidebar.tsx (hunk 1 above)
Comment (SebasthianSalpozz) [reply src/components/ContentNavigator/Sidebar.tsx]: Sure

File: src/components/Contentbar/Contentbar.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+interface ContentbarProps {
+  sections: { id: string; title: string }[];
+  onScrollToSection: (id: string) => void;
+}
+
+const styles = {
+  container: {
+    width: "20%",
+    backgroundColor: "#1a1a1a",
```
Comment (samuel-escalera-jalafund) [inline src/components/Contentbar/Contentbar.tsx]: For the bgcolor, use the colors in styleConstant, like:
color:{COLORS.GREEN}

File: src/components/Contentbar/Contentbar.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    color: "white",
```
Comment (samuel-escalera-jalafund) [inline src/components/Contentbar/Contentbar.tsx]: For the bgcolor, use the colors in styleConstant, like:
color:{COLORS.GREEN}

**Review by samuel-escalera-jalafund — CHANGES_REQUESTED** [review]

File: src/pages/OverviewDocumentation/OverviewDocumentation.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      title: "Agent Description",
+      content: (
+        <>
+          <p>
+            Agent that provides a coverage report, giving information such as:
+            covered statements uncovered statements, methods name, class name
+            for example:
+          </p>
+          <img
+            src="src/assets/images/overview/Overview 1.png"
```
Comment (fabian-romero-jalafund) [inline src/pages/OverviewDocumentation/OverviewDocumentation.tsx]: import the images, do not put the path as string
Example:
import image from "../../assets/images/welcome-background.png"
src={image}

**Review by fabian-romero-jalafund — CHANGES_REQUESTED** [review]

**Review by fabian-romero-jalafund — APPROVED** [review]

**Review by samuel-escalera-jalafund — APPROVED** [review]

**Pull request merged**